src/app/server/Server.cpp

```diff
@@ -518,6 +520,20 @@ void Server::RejoinExistingMulticastGroups()
     }
 }
 
+#if CHIP_CONFIG_ENABLE_ICD_CIP
+bool Server::CheckInWouldBeSentAtBootVerifier(FabricIndex aFabricIndex, NodeId subjectID)
+{
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+    // At least one registration has a persisted entry. Do not send Check-In message.
+    // This is to cover the use-case where the subscription resumption feature is used with the Check-In message.
```

Comment:
See comments above about what this should maybe say.

Reply:
renamed